### PR TITLE
DOCSUP-14089: Document the max_hyperscan_regexp_length and max_hyperscan_regexp_tot…

### PR DESCRIPTION
**Pull request by kirillikoff**
…al_length settings

I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Documentation for #27378

### CODE REVIEW

File: docs/en/operations/settings/settings.md [hunk 1]
```diff
@@ -3532,3 +3532,92 @@ Possible values:
 -   Positive integer.
 
 Default value: `1000`.
+
+## max_hyperscan_regexp_length {#max-hyperscan-regexp-length}
+
+The setting defines the maximum length for each regular expression in the [hyperscan multi-match functions](../../sql-reference/functions/string-search-functions.md#multimatchanyhaystack-pattern1-pattern2-patternn). If the value is `0`, the size of the regular expression is not limited.
```
Comment (olgarev) [inline docs/en/operations/settings/settings.md:3538]:
```suggestion
Defines the maximum length for each regular expression in the [hyperscan multi-match functions](../../sql-reference/functions/string-search-functions.md#multimatchanyhaystack-pattern1-pattern2-patternn). If the value is `0`, the size of the regular expression is not limited.
```

File: docs/en/operations/settings/settings.md [hunk 2]
```diff
@@ -3532,3 +3532,92 @@ Possible values:
...
+
+Possible values:
+
+-   Positive integer.
+
```
Comment (olgarev) [inline docs/en/operations/settings/settings.md:3543]: -   0 - The length is not limited.

File: docs/en/operations/settings/settings.md [hunk 3]
```diff
@@ -3532,3 +3532,92 @@ Possible values:
...
 
 Default value: `1000`.
+
+## max_hyperscan_regexp_length {#max-hyperscan-regexp-length}
+
+The setting defines the maximum length for each regular expression in the [hyperscan multi-match functions](../../sql-reference/functions/string-search-functions.md#multimatchanyhaystack-pattern1-pattern2-patternn). If the value is `0`, the size of the regular expression is not limited.
+
+Possible values:
+
+-   Positive integer.
```
Comment (olgarev) [inline docs/en/operations/settings/settings.md:3542]:
```suggestion
-   Positive integer.
-   0 - The length is not limited.
```

File: docs/en/operations/settings/settings.md [hunk 4]
```diff
@@ -3532,3 +3532,92 @@ Possible values:
...
+
+```sql
+SET max_hyperscan_regexp_length = 2;
+select multiMatchAny('abcd', ['ab','bcd','c','d']);
+```
+
+Result:
+
+```text
+Regexp length too large.
```
Comment (olgarev) [inline docs/en/operations/settings/settings.md:3573]:
```suggestion
Exception: Regexp length too large.
```

File: docs/en/operations/settings/settings.md [hunk 5]
```diff
@@ -3532,3 +3532,92 @@ Possible values:
...
+┌─multiMatchAny('abcd', ['a', 'b', 'c', 'd'])─┐
+│                                           1 │
+└─────────────────────────────────────────────┘
+```
+
+Query:
+
+```sql
+SET max_hyperscan_regexp_length = 2;
+select multiMatchAny('abcd', ['ab','bcd','c','d']);
```
Comment (olgarev) [inline docs/en/operations/settings/settings.md:3567]:
```suggestion
SELECT multiMatchAny('abcd', ['ab','bcd','c','d']);
```

File: docs/en/operations/settings/settings.md [hunk 6]
```diff
@@ -3532,3 +3532,92 @@ Possible values:
...
+
+Default value: `0`.
+
+**Example**
+
+Query:
+
+```sql
+SET max_hyperscan_regexp_length = 2;
+SELECT multiMatchAny('abcd', ['a','b','c','d']);
```
Comment (olgarev) [inline docs/en/operations/settings/settings.md:3552]:
```suggestion
SET max_hyperscan_regexp_length = 0;
SELECT multiMatchAny('abcd', ['ab','bcd','c','d']);
```

File: docs/en/operations/settings/settings.md [hunk 7]
```diff
@@ -3532,3 +3532,92 @@ Possible values:
...
+```
+
+Result:
+
+```text
+┌─multiMatchAny('abcd', ['a', 'b', 'c', 'd'])─┐
+│                                           1 │
+└─────────────────────────────────────────────┘
```
Comment (olgarev) [inline docs/en/operations/settings/settings.md:3560]:
```suggestion
┌─multiMatchAny('abcd', ['ab', 'bcd', 'c', 'd'])─┐
│                                              1 │
└────────────────────────────────────────────────┘

```

File: docs/en/operations/settings/settings.md (hunk 5 above)
Comment (olgarev) [inline docs/en/operations/settings/settings.md:3567]:
```suggestion
SELECT multiMatchAny('abcd', ['ab','bcd','c','d']) SETTINGS max_hyperscan_regexp_length = 2;
```

File: docs/en/operations/settings/settings.md [hunk 8]
```diff
@@ -3532,3 +3532,92 @@ Possible values:
...
+```
+
+**See Also**
+
+-   [max_hyperscan_regexp_total_length](#max-hyperscan-regexp-total-length)
+
+
+## max_hyperscan_regexp_total_length {#max-hyperscan-regexp-total-length}
+
+Sets the maximum length total of all regular expressions in each [hyperscan multi-match function](../../sql-reference/functions/string-search-functions.md#multimatchanyhaystack-pattern1-pattern2-patternn). If the value is `0`, the total size of the regular expressions is not limited.
```
Comment (olgarev) [inline docs/en/operations/settings/settings.md:3583]:
```suggestion
Sets the maximum length total of all regular expressions in each [hyperscan multi-match function](../../sql-reference/functions/string-search-functions.md#multimatchanyhaystack-pattern1-pattern2-patternn). 
```

File: docs/en/operations/settings/settings.md [hunk 9]
```diff
@@ -3532,3 +3532,92 @@ Possible values:
...
+
+Possible values:
+
+-   Positive integer.
```
Comment (olgarev) [inline docs/en/operations/settings/settings.md:3587]:
```suggestion
-   Positive integer.
-   0 - The length is not limited.
```

File: docs/en/operations/settings/settings.md [hunk 10]
```diff
@@ -3532,3 +3532,92 @@ Possible values:
...
+
+```sql
+SET max_hyperscan_regexp_total_length = 5;
+select multiMatchAny('abcd', ['ab','bc','c','d']);
+```
+
+Result:
+
+```text
+Total regexp lengths too large.
```
Comment (olgarev) [inline docs/en/operations/settings/settings.md:3618]:
```suggestion
Exception:  Total regexp lengths too large.
```

File: docs/en/operations/settings/settings.md [hunk 11]
```diff
@@ -3532,3 +3532,92 @@ Possible values:
...
+
+Default value: `0`.
+
+**Example**
+
+Query:
+
+```sql
+SET max_hyperscan_regexp_total_length = 5;
+SELECT multiMatchAny('abcd', ['a','b','c','d']);
```
Comment (olgarev) [inline docs/en/operations/settings/settings.md:3597]:
```suggestion
SELECT multiMatchAny('abcd', ['a','b','c','d']) SETTINGS max_hyperscan_regexp_total_length = 5;
```

File: docs/en/operations/settings/settings.md [hunk 12]
```diff
@@ -3532,3 +3532,92 @@ Possible values:
...
+┌─multiMatchAny('abcd', ['a', 'b', 'c', 'd'])─┐
+│                                           1 │
+└─────────────────────────────────────────────┘
+```
+
+Query:
+
+```sql
+SET max_hyperscan_regexp_total_length = 5;
+select multiMatchAny('abcd', ['ab','bc','c','d']);
```
Comment (olgarev) [inline docs/en/operations/settings/settings.md:3612]:
```suggestion
SELECT multiMatchAny('abcd', ['ab','bc','c','d']) SETTINGS max_hyperscan_regexp_total_length = 5;
```

File: docs/ru/operations/settings/settings.md [hunk 13]
```diff
@@ -3341,3 +3341,91 @@ SETTINGS index_granularity = 8192 │
...
+SELECT multiMatchAny('abcd', ['ab','bcd','c','d']) SETTINGS max_hyperscan_regexp_length = 2;
+```
+
+Результат:
+
+```text
+Exception: Regexp length too large.
+```
+
+**Смотрите также**
```
Comment (olgarev) [inline docs/ru/operations/settings/settings.md:3385]:
```suggestion
**См. также**
```

File: docs/ru/operations/settings/settings.md [hunk 14]
```diff
@@ -3341,3 +3341,91 @@ SETTINGS index_granularity = 8192 │
...
+SELECT multiMatchAny('abcd', ['ab','bc','c','d']) SETTINGS max_hyperscan_regexp_total_length = 5;
+```
+
+Результат:
+
+```text
+Exception: Total regexp lengths too large.
+```
+
+**Смотрите также**
```
Comment (olgarev) [inline docs/ru/operations/settings/settings.md:3429]:
```suggestion
**См. также**
```

File: docs/ru/operations/settings/settings.md [hunk 15]
```diff
@@ -3341,3 +3341,91 @@ SETTINGS index_granularity = 8192 │
 -   Положительное целое число.
 
 Значение по умолчанию: `1000`.
+
+## max_hyperscan_regexp_length {#max-hyperscan-regexp-length}
+
+Задает максимальную длину каждого регулярного выражения в [hyperscan-функциях поиска](../../sql-reference/functions/string-search-functions.md#multimatchanyhaystack-pattern1-pattern2-patternn) множественных совпадений в строке.
```
Comment (olgarev) [inline docs/ru/operations/settings/settings.md:3347]:
```suggestion
Задает максимальную длину каждого регулярного выражения в [hyperscan-функциях](../../sql-reference/functions/string-search-functions.md#multimatchanyhaystack-pattern1-pattern2-patternn)  поиска множественных совпадений в строке. 
```

File: docs/ru/operations/settings/settings.md [hunk 16]
```diff
@@ -3341,3 +3341,91 @@ SETTINGS index_granularity = 8192 │
...
+```
+
+**Смотрите также**
+
+-   [max_hyperscan_regexp_total_length](#max-hyperscan-regexp-total-length)
+
+
+## max_hyperscan_regexp_total_length {#max-hyperscan-regexp-total-length}
+
+Задает максимальную общую длину всех регулярных выражений в каждой [hyperscan-функции поиска](../../sql-reference/functions/string-search-functions.md#multimatchanyhaystack-pattern1-pattern2-patternn) множественных совпадений в строке.
```
Comment (olgarev) [inline docs/ru/operations/settings/settings.md:3392]:
```suggestion
Задает максимальную общую длину всех регулярных выражений в каждой [hyperscan-функции](../../sql-reference/functions/string-search-functions.md#multimatchanyhaystack-pattern1-pattern2-patternn)  поиска множественных совпадений в строке.
```

**Review by FArthur-cmd — APPROVED** [review]

**Pull request merged**